mithril-aggregator/src/services/signed_entity.rs

```diff
@@ -96,12 +96,22 @@ impl MithrilSignedEntityService {
             SignedEntityType::MithrilStakeDistribution(epoch) => Ok(Arc::new(
                 self.mithril_stake_distribution_artifact_builder
                     .compute_artifact(epoch, certificate)
-                    .await?,
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "Signed Entity Service can not compute artifact for epoch: '{epoch}'"
```

Comment:
Also needed for the second variant below
```suggestion
                            "Signed Entity Service can not compute artifact for entity type: '{signed_entity_type}'"
```